File: src/fragment/VoidFragment.ts
import { Text } from "pixi.js";
import Fragment, { FragmentType, register } from "./Fragment";
import { unit } from "~/endecTomfoolery";


export default class VoidFragment extends Fragment {
    static INSTANCE: VoidFragment

    private constructor() {
        super();
    }

    override toString(): string {
        return "Void";
    }

    override type(): FragmentType<VoidFragment> {
        return VOID;
    }
}

const VOID = register("void", 0x4400aa, unit(VoidFragment.INSTANCE));

File: src/jsMain/kotlin/me/maplesyrum/tricksterstudio/spell/fragment/VoidFragment.kt
package me.maplesyrum.tricksterstudio.spell.fragment

import me.maplesyrum.tricksterstudio.endec.unit
import tree.maple.kendec.StructEndec

class VoidFragment : Fragment() {
    override fun type(): FragmentType<*> {
        return FragmentType.VOID
    }

    override fun toString(): String {
        return "Void";
    }

    companion object {
        val INSTANCE: VoidFragment = VoidFragment()

        val ENDEC: StructEndec<VoidFragment> = unit(INSTANCE)
    }
}
